tests/unit_tests/provider_tests/test_ensemble_table_provider.py

```diff
@@ -0,0 +1,180 @@
+
+    df = model.get_column_data(["STR"], [1])
+    assert df.shape == (4, 2)
+    assert df.columns.tolist() == ["REAL", "STR"]
+
+
+# -------------------------------------------------------------------------
+def test_create_from_aggregated_csv_file_smry_csv(
+    testdata_folder: Path, tmp_path: Path
+) -> None:
+    factory = EnsembleTableProviderFactory(
+        tmp_path, backing_type=BACKING_TYPE_TO_TEST, allow_storage_writes=True
+    )
+    providerset = factory.create_provider_set_from_aggregated_csv_file(
+        testdata_folder / "aggregated_data" / "smry.csv"
+    )
+
+    assert providerset.ensemble_names() == ["iter-0"]
+    provider = providerset.ensemble_provider("iter-0")
+
+    assert len(provider.column_names()) == 17
+    assert provider.column_names()[0] == "DATE"
+    assert provider.column_names()[16] == "YEARS"
+
+    assert len(provider.realizations()) == 40
+
+    valdf = provider.get_column_data(["YEARS"])
+    assert len(valdf.columns) == 2
+    assert valdf.columns[0] == "REAL"
+    assert valdf.columns[1] == "YEARS"
+    assert valdf["REAL"].nunique() == 40
+
+    valdf = provider.get_column_data(["YEARS"], [0, 39, 10])
+    assert len(valdf.columns) == 2
+    assert valdf.columns[0] == "REAL"
+    assert valdf.columns[1] == "YEARS"
+    assert valdf["REAL"].nunique() == 3
+
+
+# -------------------------------------------------------------------------
+def test_create_from_aggregated_csv_file_smry_hm(
+    testdata_folder: Path, tmp_path: Path
+) -> None:
+    factory = EnsembleTableProviderFactory(
+        tmp_path, backing_type=BACKING_TYPE_TO_TEST, allow_storage_writes=True
+    )
+    providerset = factory.create_provider_set_from_aggregated_csv_file(
+        testdata_folder / "aggregated_data" / "smry_hm.csv"
+    )
+
+    assert providerset.ensemble_names() == ["iter-0", "iter-3"]
+    provider = providerset.ensemble_provider("iter-0")
+
+    assert len(provider.column_names()) == 474
+    assert provider.column_names()[0] == "DATE"
+    assert provider.column_names()[1] == "BPR:15,28,1"
+    assert provider.column_names()[473] == "YEARS"
+
+    assert len(provider.realizations()) == 10
+
+    valdf = provider.get_column_data(["DATE"])
+    assert len(valdf.columns) == 2
+    assert valdf.columns[0] == "REAL"
+    assert valdf.columns[1] == "DATE"
+    assert valdf["REAL"].nunique() == 10
+
+
+# -------------------------------------------------------------------------
+def test_create_from_per_realization_csv_file(
+    testdata_folder: Path, tmp_path: Path
+) -> None:
+
+    ensembles: Dict[str, str] = {
+        "iter-0": str(testdata_folder / "reek_history_match/realization-*/iter-0"),
+        "iter-1": str(testdata_folder / "reek_history_match/realization-*/iter-1"),
+        "iter-2": str(testdata_folder / "reek_history_match/realization-*/iter-2"),
+        "iter-3": str(testdata_folder / "reek_history_match/realization-*/iter-3"),
+    }
+
+    csvfile = "share/results/tables/rft.csv"
```

Comment:
I guess we already have code like this, but this would fail on Windows due to `".../..."`.

Reply:
Should we address this now?

Reply:
It can wait :+1: (line 144-147 could easily be made Windows compatible now though, but it will not work anyway due to the existing `/`).
```suggestion
    ensembles: Dict[str, str] = {
        "iter-0": str(testdata_folder / "reek_history_match" / "realization-*" / "iter-0"),
        "iter-1": str(testdata_folder / "reek_history_match" / "realization-*" / "iter-1"),
        "iter-2": str(testdata_folder / "reek_history_match" / "realization-*" / "iter-2"),
        "iter-3": str(testdata_folder / "reek_history_match" / "realization-*" / "iter-3"),
    }

    csvfile = "share/results/tables/rft.csv"
```